src/text-fragment-utils.js

```diff
@@ -238,13 +345,14 @@ export const scrollElementIntoView = (element) => {
 
 /**
  * Finds the DOM Node and the exact offset where a string starts or ends.
- * @param {Node} blockNode - Block element in which to search for a given text.
+ * @param {HTMLElement} blockNode - Block element in which to search for a given text.
  * @param {string} text - The text for which to find the position.
  * @param {boolean} start - Whether to return the an offset for the start or the text, or the end.
  * @return {[Node, number]} The DOM Node and the offset where the text starts or ends.
  */
 const findRangeNodeAndOffset = (blockNode, text, start) => {
-  let offset = blockNode.textContent.indexOf(text) + (start ? 0 : text.length);
+  const fullText = blockNode.innerText.replace(/\s/g, ' ');
```

Comment:
I wonder if we can get away without this. Shouldn't the URL encode the same kind of space character as the node's text?